core/src/main/scala/cats/data/StateT.scala

```diff
@@ -12,25 +12,32 @@ import cats.syntax.either._
 final class StateT[F[_], S, A](val runF: F[S => F[(S, A)]]) extends Serializable {
 
   def flatMap[B](fas: A => StateT[F, S, B])(implicit F: Monad[F]): StateT[F, S, B] =
```

Comment:
I think that makes sense for now. Maybe add a comment to that effect? If we ever want to tighten things up later that will be handy.